samples/Web/deploy-webapp-ip-restrictions/azurepolicy.json

```diff
@@ -0,0 +1,75 @@
+{
+    "properties": {
+        "displayName": "Web App default IP restrictions",
+        "mode": "indexed",
+        "description": "Configures Web App IP restriction to allow access from specified public IP addresses",
+        "metadata": {
+            "category": "Web"
+        },
+        "parameters": {},
+        "policyRule": {
+            "if": {
+                "field": "type",
+                "equals": "Microsoft.Web/sites"
+            },
+            "then": {
+                "effect": "DeployIfNotExists",
+                "details": {
+                    "type": "Microsoft.Web/sites/config",
+                    "roleDefinitionIds": [
+                        "/providers/Microsoft.Authorization/roleDefinitions/de139f84-1756-47ae-9be6-808fbbe84772"
+                    ],
+                    "existenceCondition": {
+                        "not": {
+                            "field": "Microsoft.Web/sites/config/web.ipSecurityRestrictions[*].action",
+                            "Equals": "Allow"
```

Comment:
This existence condition doesn't look at IPs and will only be satisfied if at least one of the actions is not "Allow".  The deployment below creates two "Allow" actions which would not make this policy compliant.

Reply:
The existence condition states that all actions must be Allow, according to [this doc](https://docs.microsoft.com/en-us/azure/governance/policy/how-to/author-policies-for-arrays#evaluating-the--alias) and verified by observation.

When App Service has no rules it assumes an implicit "Allow All".  
This policy looks for this state and applies Allow rules.  
When Allow rules are present, App Service assumes an implicit "Deny All" rule at the lowest priority (highest priority number value).